src/infra/db/mssql/util/knex/extensions/turbo-interceptor.ts

```diff
@@ -0,0 +1,165 @@
+import k, { type Knex } from 'knex';
+import NodeCache from 'node-cache';
+
+import { logger } from '@/util';
+import { makeCacheServer } from '@/infra/cache';
+
+type Services = 'memjs' | 'node-cache';
+type GenericObject = Record<string, unknown>;
+
+const cache = new NodeCache({ stdTTL: 100, checkperiod: 120 });
+const memCache = makeCacheServer();
+
+const MAX_MEMORY_USAGE = 100 * 1024 * 1024; // 100 MB
+const MAX_RESULT_SIZE = 1024 * 1024; // 1 MB
+
+function checkMemoryUsage() {
+  const memoryUsage = process.memoryUsage();
+  return memoryUsage.heapUsed < MAX_MEMORY_USAGE;
+}
+
+function cacheKey(sql: Knex.Sql) {
+  return sql.sql + JSON.stringify(sql.bindings);
+}
+
+function nodeCacheQuery(sql: Knex.Sql, result: unknown) {
+  const key = cacheKey(sql);
+  cache.set(key, result);
+}
+
+async function memCacheQuery(sql: Knex.Sql, result: GenericObject) {
+  await memCache.set({ key: cacheKey(sql), value: result });
+}
+
+async function cacheQuery(sql: Knex.Sql, result: unknown, service: Services) {
+  if (service === 'node-cache') {
+    return nodeCacheQuery(sql, result);
+  }
+
+  return memCacheQuery(sql, <GenericObject>result);
+}
+
+async function getCachedQueryMemCache(sql: Knex.Sql) {
+  const key = cacheKey(sql);
+  const buffer = await memCache.get(key);
+
+  if (!buffer) return;
+
+  try {
+    return JSON.parse(buffer.toString());
+  } catch (error) {
+    return buffer.toString();
+  }
+}
+
+function getCachedQueryNodeCache(sql: Knex.Sql) {
+  const key = cacheKey(sql);
+  return cache.get(key);
+}
+
+async function getCachedQuery(sql: Knex.Sql, service: Services) {
+  if (service === 'node-cache') return getCachedQueryNodeCache(sql);
+  return getCachedQueryMemCache(sql);
+}
+
+function saveToCache(sql: Knex.Sql, result: unknown, service: Services) {
+  setImmediate(async () => {
+    const resultSize = Buffer.byteLength(JSON.stringify(result));
+
+    if (resultSize > MAX_RESULT_SIZE) {
+      logger.log({
+        level: 'warn',
+        message: `Result size ${resultSize} exceeds maximum allowed size of ${MAX_RESULT_SIZE}. Cache not saved.`
+      });
+      return;
+    }
+
+    await cacheQuery(sql, result, service);
+  });
+}
+
+function resolveWrapperToSaveInCache(
+  resolve: (data: Object | Object[]) => void,
+  sql: Knex.Sql,
+  service: Services
+) {
+  return (data: Object | Object[]) => {
+    try {
+      if (typeof data !== 'object') {
+        resolve(data);
+        return;
+      }
+
+      if (!checkMemoryUsage()) {
+        logger.log({
+          level: 'warn',
+          message: 'Memory usage exceeded maximum allowed. Cache not saved.'
+        });
+        return resolve(data);
+      }
+
+      saveToCache(sql, data, service);
+
+      resolve(data);
+    } catch (error) {
```

Comment:
Rename error to _error.

This is okay, I suspect this catch will never be triggered, but it's better to be safe...